integration-tests/test_tags.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+import conftest
+import contextlib
+import os
+import yaml
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_tags(insights_client, external_inventory, test_config):
+    """
+    steps:
+        1. Register insights-client, and check satellite related
+           tags on the inventory if the test env is satellte.
+
+        2. Run insights-client with the --group
+
+        3. Add a new tag in tags.yaml, and run insights-client,
+           then check the inventory
+    :expectedresults:
+        1. system is registered to insights, and there will be
+           satellite related tags supported by branch_info with
+           satellite env.
+
+        2. tags.yaml will be created with group option, and new tag
+           generated by tags.yaml
+
+        3. The new tag shows on inventory by modifying tags.yaml
+    """
+    # Register insights
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    # Remove the tags.yaml if it exists
+    with contextlib.suppress(FileNotFoundError):
+        os.remove("/etc/insights-client/tags.yaml")
```

Comment:
Should this happen _before_ the registration? If some test left the file there, the only-satellite-tags below assert would fail.